cirq-core/cirq/value/linear_dict.py

```diff
@@ -34,7 +34,7 @@
 )
 from typing_extensions import Self
 
-Scalar = Union[complex, float, numbers.Complex]
+Scalar = complex
```

Comment:
Does this still cover floats and numpy complex numbers?

Reply:
Good catch.  `int` and `float` are OK for `complex` annotation per [PEP-0484](https://peps.python.org/pep-0484/#the-numeric-tower), but numpy types did not match.  Turns out they also did not match in the main branch (2f922cdd74862c586a8bda1f8f59cb7c73d347d1) with mypy-1.2.0, but it looks all is good after 00765f0f51f84d05e1c158b9dd01045f1ff7ce2b here.

Here is a test to check if numeric types pass -
```python
# linear_dict_typing.py -- check with
# mypy --config-file=dev_tools/conf/mypy.ini linear_dict_typing.py

import numpy as np
import cirq

cirq.LinearDict({'a': 1})
cirq.LinearDict({'a': np.int32(32)})
cirq.LinearDict({'a': np.int64(64)})

cirq.LinearDict({'a': 2.0})
cirq.LinearDict({'a': np.float32(32.0)})
cirq.LinearDict({'a': np.float64(64.0)})

cirq.LinearDict({'a': complex(3j)})
cirq.LinearDict({'a': np.complex64(3j)})
cirq.LinearDict({'a': np.complex128(3j)})
```